src/configuration.js

```diff
@@ -80,58 +54,58 @@ export default class Configuration {
         }
 
         if (type === 'remote') {
-            sampler = new RemoteSampler(config.serviceName, {
+            sampler = new RemoteSampler(serviceName, {
                 sampler: new ProbabilisticSampler(param)
             });
         }
 
+        if (!sampler) {
+            sampler = new ProbabilisticSampler(0.001);
+        }
+
         return sampler;
     }
 
-    static _getReporter(config, options) {
-        let sender;
-        let reporterConfig = {};
-        let reporters = [];
-        let hostPort = '';
-        if (config.reporter) {
-            if (config.reporter.logSpans) {
+    static _getReporter(config: ReporterConfig, options: any): Reporter {
+        let sender: Sender;
+        let reporterConfig: any = {};
+        let reporters: Array<Reporter> = [];
+        let hostPort: string = '';
+        if (config) {
+            if (config.logSpans) {
                 reporters.push(new LoggingReporter(options.logger));
             }
 
-            if (config.reporter.flushIntervalMs) {
-                reporterConfig['bufferFlushInterval'] = config.reporter.flushIntervalMs;
+            if (config.flushIntervalMs) {
+                reporterConfig['flushIntervalMs'] = config.flushIntervalMs;
             }
 
-            if (config.reporter.agentHost && config.reporter.agentPort) {
-                hostPort = `${config.reporter.agentHost}:${config.reporter.agentPort}`;
+            if (config.agentHost && config.agentPort) {
+                hostPort = `${config.agentHost}:${config.agentPort}`;
             }
         }
 
+        // $FlowIgnore - disable type inference for udpsender.
         sender = new UDPSender(hostPort);
         reporters.push(new RemoteReporter(sender, reporterConfig));
         return new CompositeReporter(reporters);
     }
 
-    static initTracer(config, options = {}) {
-        let v = new Validator();
-        v.validate(config, jaegerSchema, {
-            throwError: true
-        });
-
-        let reporters = [];
-        let reporter;
-        let sampler;
+    static initTracer(config: TracerConfig, options: any = {}): Tracer {
```

Comment:
* please document `options`. Maybe add another type to `src/decls/types.js`? Or even better, why can we not use just the TracerConfig for that?  That's what we do in other languages - some fields of the config object are meant to be loaded from a file, but others can be assigned programmatically, like the logger.
* add comment `// see ./decls/types.js for TracerConfig`